### PR TITLE
[11.13] Addition of Status & Environment to list of possible values

### LINKED ISSUE #768: Add support for additional filters in the Deployment API
In https://docs.gitlab.com/ee/api/deployments.html there are additional filters available and I'd like to use at least a few of them...  

I'll submit a MR shortly with the updates I think are needed...

### PR DESCRIPTION
**Pull request by donkidd**
Added Status an Environment as parameters that can be sent to deployment method.  This is to fix #768

### CODE REVIEW

File: src/Api/Deployments.php [hunk 1]
```diff
@@ -36,6 +36,11 @@ public function all($project_id, array $parameters = [])
         $resolver->setDefined('sort')
             ->setAllowedTypes('sort', 'string')
             ->setAllowedValues('sort', ['desc', 'asc']);
+        $resolver->setDefined('status')
```
Comment (GrahamCampbell) [inline src/Api/Deployments.php:43]: Please also update the phpdoc

**Review by GrahamCampbell — CHANGES_REQUESTED** [review]
Thanks for the PR. I've left notes inline to be addressed before merge.

File: src/Api/Deployments.php (hunk 1 above)
Comment (donkidd) [reply src/Api/Deployments.php:43]: Ohh yea forgot about that, have updated the phpdoc with new values

Comment (donkidd) [reply src/Api/Deployments.php:43]: Finally got Style to pass, I kept missing a space in my phpdoc...

**Comment by GrahamCampbell** [comment]
r.e. StyleCI, it gives you the diff with the fixes to download and then apply using git, so you don't have to take multiple tries. ;)

---

image

https://github.styleci.io/analyses/YOdYGj/diff

**Comment by GrahamCampbell** [comment]
Thanks.

**Pull request merged**